### PR TITLE
macho: implement support for bundles

### PR DESCRIPTION
**Pull request by yrashk**
MachO linker now handles `-bundle` and `-bundle_loader <executable>` parameters similar to how `ld` does it.

There is no test case for this proof of concept just yet but I am planning to prepare one.

@kubkon what do you think?

### CODE REVIEW

File: src/Compilation.zig [hunk 1]
```diff
@@ -185,6 +185,9 @@ astgen_wait_group: WaitGroup = .{},
 /// TODO: Remove this when Stage2 becomes the default compiler as it will already have this information.
 export_symbol_names: std.ArrayListUnmanaged([]const u8) = .{},
 
+bundle: bool = false,
```
Comment (kubkon) [inline src/Compilation.zig:188]: Are these needed at this scope? I would think that having them in `InitOptions` should be enough.

File: src/Compilation.zig [hunk 2]
```diff
@@ -1036,6 +1039,8 @@ pub const InitOptions = struct {
     /// (Darwin) remove dylibs that are unreachable by the entry point or exported symbols
     dead_strip_dylibs: bool = false,
     libcxx_abi_version: libcxx.AbiVersion = libcxx.AbiVersion.default,
+    bundle: bool = false,
```
Comment (kubkon) [inline src/Compilation.zig:1042]: Please add doc-style comments like `/// (Darwin) ...`.

**Review by kubkon — CHANGES_REQUESTED** [review]
Thanks for the PR! I have left you some basic comments in the code, but the major missing bit is a matching linker test. You can find existing linker tests in [test/link.zig](https://github.com/ziglang/zig/blob/master/test/link.zig). They are all `build.zig`-based, like our `standalone` tests so with your additions, it should be very easy to add a test for `MH_BUNDLE`.

In the meantime, I will need to understand the process of creating `MH_BUNDLE`s to be able to offer a more thorough review and get it into a mergeable state.

File: src/Compilation.zig (hunk 1 above)
Comment (yrashk) [reply src/Compilation.zig]: Removed.

File: src/Compilation.zig (hunk 2 above)
Comment (yrashk) [reply src/Compilation.zig:1040]: Done.

**Comment by yrashk** [comment]
I am working on a test case.

**Comment by kubkon** [comment]
Just wanted to let you know that I very much appreciate your help on this, and will get back to reviewing it and helping you in getting it landed as soon as I finish implementing synthesising unwind records for MachO linker, so take your time on the test case.

**Comment by yrashk** [comment]
It'll take time, it looks like I didn't quite understand how the linker works yet so it seemingly did not produce the right file yet. Maybe a quick pair session will be useful here if you're interested (when you're available)

**Comment by kubkon** [comment]
> It'll take time, it looks like I didn't quite understand how the linker works yet so it seemingly did not produce the right file yet. Maybe a quick pair session will be useful here if you're interested (when you're available)

Sounds good although as I mentioned above it will have to wait until I make some good headway on the unwind records. Let's sync up on discord!

**Comment by kubkon** [comment]
@yrashk, thanks for working on this.

This PR is not mergeable as-is because more research is needed into the `MH_BUNDLE` use-case by me so that we properly integrate it into the linker. Unfortunately I have other priorities at the moment, so I will not be working on it until a later time.

As for next steps, I would suggest opening an issue to track this missing feature and link this PR so that I will revisit this when I have more time.

**Pull request closed without merging**